### PR TITLE
Add a benchmark for simple DisplayLists to exercise the Raster Cache's cache admission algorithm

### PR DESCRIPTION
**Pull request by gw280**
This is an initial benchmark that allows us to track the status quo for a simple DisplayList that is cheap to rasterise but still gets cached anyway.

https://github.com/flutter/flutter/issues/86728

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: .ci.yaml [hunk 1]
```diff
@@ -2084,6 +2084,16 @@ targets:
       task_name: picture_cache_perf__timeline_summary
     scheduler: luci
 
+  - name: Linux_android picture_cache_complexity_scoring_perf__timeline_summary
```
Comment (keyonghan) [inline .ci.yaml:2087]: Could we add `bringup: true` for both tests? Flakiness needs to get validated in staging pool first before enabled in prod.

**Review by CaseyHillers — APPROVED** [review]
ci.yaml lgtm

Edit: See @keyonghan comment about bringup

File: .ci.yaml (hunk 1 above)
Comment (flar) [inline .ci.yaml:2087]: Add `    bringup: true` as one of the first two property lines on new benchmarks. The bots will monitor the flaky status and automatically submit a deflaking PR when the consistency is established.

File: .ci.yaml [hunk 2]
```diff
@@ -3391,6 +3401,16 @@ targets:
       task_name: new_gallery_ios__transition_perf
     scheduler: luci
 
+  - name: Mac_ios picture_cache_complexity_scoring_perf__timeline_summary
```
Comment (flar) [inline .ci.yaml:3405]: bringup: true

File: .ci.yaml (hunk 1 above)
Comment (flar) [reply .ci.yaml:2087]: (I see @keyonghan already mentioned this while I was writing my comments.)

File: dev/benchmarks/macrobenchmarks/test_driver/picture_cache_complexity_scoring_perf_test.dart [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+import 'util.dart';
+
+void main() {
+  macroPerfTest(
+    'picture_cache_complexity_scoring_perf',
+    kPictureCacheComplexityScoringRouteName,
+    pageDelay: const Duration(seconds: 1),
+    driverOps: (FlutterDriver driver) async {
+      final SerializableFinder tabBarView = find.byValueKey('tabbar_view');
```
Comment (flar) [inline dev/benchmarks/macrobenchmarks/test_driver/picture_cache_complexity_scoring_perf_test.dart:16]: Should this have a unique name so as not to be confused with the tbv used in the picture_cache benchmark you copied this from?

**Review by flar — APPROVED** [review]

**Review by keyonghan — APPROVED** [review]
LGTM re .ci.yaml change.

File: dev/benchmarks/macrobenchmarks/test_driver/picture_cache_complexity_scoring_perf_test.dart (hunk 3 above)
Comment (flar) [reply dev/benchmarks/macrobenchmarks/test_driver/picture_cache_complexity_scoring_perf_test.dart:16]: You changed the Key, but not the Finder...?

Comment (gw280) [reply dev/benchmarks/macrobenchmarks/test_driver/picture_cache_complexity_scoring_perf_test.dart]: I fixed it, accidentally edited it in the wrong file...

**Pull request merged**